compiler/compiler.go

```diff
@@ -72,15 +72,15 @@ func Compile(scope Scope, f *semantic.FunctionExpression, in semantic.MonoType)
 // inType and mapping any variables to the value in the other record.
 // If the input type is not a type variable, it will check to ensure
 // that the type in the input matches or it will return an error.
-func substituteTypes(subst map[uint64]semantic.MonoType, inType, in semantic.MonoType) error {
+func substituteTypes(subst map[uint64]semantic.MonoType, inferredType, actualType semantic.MonoType) error {
```

Comment:
I changed the name of the arguments here just to help make the code clearer.

Reply:
Thank you for this.